### PR TITLE
Guard against NRE in Message.cs

### PR DESCRIPTION
**Pull request by drwill-ms**
Fix for bug [bug 998](https://github.com/Azure/iotedge/issues/998)

Properties and SystemProperties values can be null, so do not deref them blindly.

### CODE REVIEW

**Comment by drwill-ms** [comment]
I hit PR # 1000. What do I win?

File: edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/MessageTest.cs [hunk 1]
```diff
@@ -72,15 +76,25 @@ public void TestSet()
...
             Assert.Contains(Message3, messages);
             Assert.Contains(Message4, messages);
             Assert.Contains(Message6, messages);
+            Assert.Contains(Message9, messages);
             Assert.DoesNotContain(Message5, messages);
         }
+
+        [Fact]
+        [Unit]
+        public void MessageWithNullValueInPropertiesShouldNotThrow()
```
Comment (varunpuranik) [inline edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/MessageTest.cs]: Nit: can we rename this to something like MessageSizeWithNullPropertyValuesShouldNotThrow.. ?

Comment (drwill-ms) [reply edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/MessageTest.cs]: Sure, no problem.

**Comment by damonbarry** [comment]
🎉🎈🎉🎈

**Pull request merged**